govcd/org.go

```diff
@@ -235,18 +237,26 @@ func (org *Org) GetCatalogByName(catalogName string, refresh bool) (*Catalog, er
 // On success, returns a pointer to the Catalog structure and a nil error
 // On failure, returns a nil pointer and an error
 func (org *Org) GetCatalogById(catalogId string, refresh bool) (*Catalog, error) {
-	if refresh {
-		err := org.Refresh()
+	// ignoring error as the Uuid might be invalid when GetCatalogByNameOrId is used
+	bareCatalogId, _ := getBareEntityUuid(catalogId)
+
+	catalogs, err := org.QueryCatalogList()
```

Comment:
it would be faster/easier to query using ID in filter

Reply:
It makes sense and I changed it.